pkg/build/build_implementation.go

```diff
@@ -141,39 +141,40 @@ func (bc *Context) buildImage(ctx context.Context) ([]*apk.Package, error) {
 		// Index for loop to make golang-ci happy.
 		// See https://stackoverflow.com/questions/62446118/implicit-memory-aliasing-in-for-loop
 		for index := range basePkgs {
-			err := bc.apk.AddInstalledPackage(&basePkgs[index].Package, basePkgs[index].Files)
+			_, err := bc.apk.AddInstalledPackage(&basePkgs[index].Package, basePkgs[index].Files)
 			if err != nil {
-				return nil, err
+				return nil, nil, err
 			}
 		}
 	}
 
 	var (
-		pkgs []*apk.Package
-		err  error
+		pkgs  []*apk.Package
+		diffs [][]byte
```

Comment:
This is only ever going to be an additive diff right? This is "each layer's installed DB", and the solve that has already previously happened, so it won't ever be removing anything, only splitting the adds into chunks.

I don't know if this makes sense, or if there's some better word for "diff" that might make it clearer to read.

(Not blocking at all, just musing)